src/floppy.js

```diff
@@ -409,6 +452,12 @@ FloppyController.prototype.done = function(args, cylinder, head, sector, error)
         }
     }
 
+    // clear eject flag if seek or write has taken us to a new cylinder
+    if(cylinder != this.last_cylinder) {
+        this.dir = 0x80;
```

Comment:
This is the other location @copy mentioned (just adding a comment here so I can track it more easily).